lib/Dialect/CGGI/Transforms/BooleanVectorizer.cpp

```diff
@@ -139,36 +142,58 @@ int bucketSize(const SmallVector<SmallVector<Operation *>> &buckets) {
   return size;
 }
 
+int countNotOps(DenseMap<Operation *, SmallVector<SmallVector<Operation *>>>
+                    compatibleOps) {
+  int count = 0;
+  DenseMap<Operation *, SmallVector<SmallVector<Operation *>>>::iterator it;
+
+  for (it = compatibleOps.begin(); it != compatibleOps.end(); it++) {
+    if (isa<cggi::NotOp>(it->first)) count++;
+  }
```

Comment:
I would try using `llvm::count_if` if possible, then this entire function might be small enough to inline at the sole call site.

https://github.com/llvm/llvm-project/blob/197e0125c3ed991f11d97d27e529c02af06e9f37/llvm/include/llvm/ADT/STLExtras.h#L1945

Reply:
Changed it. 
Cool didn't know that was a thing!